tldw_Server_API/app/core/MCP_unified/modules/implementations/governance_module.py

```
@@ -180,19 +174,91 @@ def _optional_str(value: Any) -> str | None:
+    @staticmethod
+    def _merged_metadata(caller_metadata: dict[str, Any], authenticated_metadata: dict[str, Any]) -> dict[str, Any]:
+        merged = dict(authenticated_metadata)
+        merged.update(caller_metadata)
+        return merged
+
+    @staticmethod
+    def _overlay_scope(metadata: dict[str, Any], scope: dict[str, Any]) -> dict[str, Any]:
+        merged = dict(metadata)
+        for key, value in scope.items():
+            if value is not None:
+                merged[key] = value
```

Ruff (0.15.9)

[warning] Use a dictionary comprehension instead of a for-loop

(PERF403)